modules/nf-core/eggnogmapper/main.nf

```diff
@@ -23,11 +23,13 @@ process EGGNOGMAPPER {
     task.ext.when == null || task.ext.when
 
     script:
-    def args = task.ext.args ?: ''
-    def prefix = task.ext.prefix ?: "${meta.id}"
-    def is_compressed = fasta.name.endsWith(".gz")
-    def fasta_name = fasta.name.replace(".gz", "")
-    def dbmem = task.memory.toMega() > 40000 ? '--dbmem' : ''
+    def args            = task.ext.args                 ?: ''
+    def prefix          = task.ext.prefix               ?: "${meta.id}"
+    def is_compressed   = fasta.name.endsWith(".gz")
+    def fasta_name      = fasta.name.replace(".gz", "")
```

Comment:
Since you are at it, can you replace the is_compressed check and assignment with this?:
def is_compressed = fasta.getExtension() == "gz" ? true : false
def fasta_name = is_compressed ? fasta.getBaseName() : fasta
    
   Similarly to the blast/blastp module for example; more robust.

Reply:
Thank you @vagkaratzas 

You are right. The checks only cover empty inputs such as []. Now I have added a test to verify this case.

I have also made changes to `is_compressed` and `fasta_name`.